### PR TITLE
Migrate the `i686::sse` module to vendor types

### PR DESCRIPTION
**Pull request by alexcrichton**
This migrates the entire `i686::sse` module (and touches a few others) to the
vendor types.

### CODE REVIEW

File: coresimd/src/x86/i686/sse.rs [hunk 1]
```diff
@@ -217,57 +215,57 @@ pub unsafe fn _mm_cvtpi32_ps(a: f32x4, b: i32x2) -> f32x4 {
 #[inline(always)]
 #[target_feature = "+sse"]
 #[cfg_attr(test, assert_instr(cvtpi2ps))]
-pub unsafe fn _mm_cvt_pi2ps(a: f32x4, b: i32x2) -> f32x4 {
+pub unsafe fn _mm_cvt_pi2ps(a: __m128, b: i32x2) -> __m128 {
```
Comment (gnzlbg) [inline coresimd/src/x86/i686/sse.rs]: Why is `i32x2` still used here? Or is this only a partial migration of the `i686::sse` module?

File: coresimd/src/x86/i586/avx.rs [hunk 2]
```diff
@@ -2558,12 +2554,21 @@ mod tests {
...
+    unsafe fn assert_eq_m128(a: __m128, b: __m128) {
+        let r = _mm_cmpeq_ps(a, b);
+        if _mm_movemask_ps(r) != 0b1111 {
+            panic!("{:?} != {:?}", a, b);
+        }
+    }
+
     #[simd_test = "avx"]
-    unsafe fn _mm256_add_pd() {
+    unsafe fn test_mm256_add_pd() {
```
Comment (gnzlbg) [inline coresimd/src/x86/i586/avx.rs]: Curiosity: why these renames?

File: coresimd/src/x86/i586/avx.rs [hunk 3]
```diff
@@ -4050,66 +4055,66 @@ mod tests {
...
         assert_eq!(r, f64x4::new(1., 2., 3., 4.));
     }
 
     #[simd_test = "avx"]
-    unsafe fn _mm256_castps256_ps128() {
+    unsafe fn test_mm256_castps256_ps128() {
         let a = f32x8::new(1., 2., 3., 4., 5., 6., 7., 8.);
-        let r = avx::_mm256_castps256_ps128(a);
-        assert_eq!(r, f32x4::new(1., 2., 3., 4.));
+        let r = _mm256_castps256_ps128(a);
```
Comment (gnzlbg) [inline coresimd/src/x86/i586/avx.rs]: Why this change?

Comment (gnzlbg) [reply coresimd/src/x86/i586/avx.rs]: I guess this is why the renames are necessary.

File: coresimd/src/x86/i686/sse.rs (hunk 1 above)
Comment (gnzlbg) [reply coresimd/src/x86/i686/sse.rs]: It's the `i586::sse` module that was migrated, not the `i686::sse` one.

Comment (alexcrichton) [reply coresimd/src/x86/i686/sse.rs]: Ah yeah oops those are some typos in the PR title/description. I'll be fixing up the other modules as I go along (those one was focused on i586)
